x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java

```diff
@@ -282,6 +305,12 @@ public String toString() {
             + profile
             + ", tables="
             + tables
+            + "allow_partial_result="
+            + allowPartialResults
             + '}';
     }
+
+    public static boolean supportPartialResults(TransportVersion transportVersion) {
```

Comment:
Why do we use a separate method for this? All the other code seems to use direct conditionals.

Reply:
I prepared for backport, but I have inlined it in https://github.com/elastic/elasticsearch/pull/121942/commits/ea784cdd54962a6201f74728f0a9f8ad291e6887